### PR TITLE
**[DO NOT MERGE (still in progress)]** Missing Videos Bug

### PR DESCRIPTION
**Pull request by nathangthomas**
This pull request includes two different solutions to the user story. If you have time before Monday I would like to hear your thoughts. 

Solution_1: Add if/else statement to the tutorial's show page that alerts user if there is no video for the tutorial. I personally think this is a better solution to the problem than the recommended "Video.new" that Alex suggested in the project board because it is straight and to the point. 

Solution_2: (This one currently overrides the above solution. We probably should decide on one and delete the other) Added Video.new to the current_video method inside of the tutorial_facade so that if current_video is nil there is a default video. I was unsuccessful in getting the thumbnail for this default video to render. 

I was confused as to how I would test for this so there is an impartial test I used to hit my debuggers. With some feedback I hope to write a proper test for this.

### CODE REVIEW

File: app/views/tutorials/show.html.erb [hunk 1]
```diff
@@ -1,69 +1,74 @@
...
-    <% @facade.videos.each do |video| %>
-      <li>
-        <h3><%= link_to video.title, tutorial_path(video_id: video.id), class: "show-link", id: video.position %></h3>
-      </li>
-    <% end %>
-  </ul>
-</div>
-
-<div class="col col-8">
-  <div class="title-bookmark">
```
Comment (nancylee713) [inline app/views/tutorials/show.html.erb]: A user can still see the bookmark button when there are no videos; a minor issue, but do you think it's necessary to include a conditional statement here?

File: app/facades/tutorial_facade.rb [hunk 2]
```diff
@@ -5,7 +5,10 @@ def initialize(tutorial, video_id = nil)
   end
 
   def current_video
-    if @video_id
+    if videos == []
```
Comment (nancylee713) [inline app/facades/tutorial_facade.rb]: We could also use a ruby method like, `if videos.empty?`

File: spec/features/user/tutorial_show_page_bug_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#This test is solely for the purpose of hitting my debugger to pry into this feature.
```
Comment (nancylee713) [inline spec/features/user/tutorial_show_page_bug_spec.rb]: Yeah, currently the seed file does not contain a tutorial without videos. We could add one and see if the feature works in development.

File: spec/features/user/tutorial_show_page_bug_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    VCR.turn_off!
+
+    user = create(:user)
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    Tutorial.create!(title: 'Sample Tutorial', description: 'this is a test description')
+  end
+
+  # If a tutorial doesn't have any videos and we visit its show page an error occurs when trying to call current_video.title since current_video is nil.
+  it 'when i visit a toutorial show page i can confirm there is an error when trying to visit current_video.title when a tutorial has no videos' do
```
Comment (fentontaylor) [inline spec/features/user/tutorial_show_page_bug_spec.rb]: typo: toutorial -> tutorial

File: app/facades/tutorial_facade.rb (hunk 2 above)
Comment (fentontaylor) [reply app/facades/tutorial_facade.rb]: I was going to say the same thing

Comment (nathangthomas) [reply app/facades/tutorial_facade.rb]: Good point. I'll use that.

File: app/views/tutorials/show.html.erb (hunk 1 above)
Comment (nathangthomas) [reply app/views/tutorials/show.html.erb]: Good catch. We can just implement the ` if videos.empty?`  block inside of the tutorial_facade as a replacement for this entire conditional statement. That will solve it.

File: spec/features/user/tutorial_show_page_bug_spec.rb (hunk 4 above)
Comment (nathangthomas) [reply spec/features/user/tutorial_show_page_bug_spec.rb]: Thanks!  I'll rewrite the test once I update our seeds file to include a tutorial without videos.

**Comment by nathangthomas** [comment]
I made some changes to the previous pull request. 
-Made changes we discussed
-Friends sections on user show page no longer renders if user has no friends

**Pull request merged**